lldb/source/Plugins/Language/Swift/FoundationValueTypes.cpp

```diff
@@ -96,13 +96,52 @@ bool lldb_private::formatters::swift::URL_SummaryProvider(
     return false;
 
   std::string summary;
-  if (!underlying_url_sp->GetSummaryAsCString(summary, options))
+  auto try_format = [&]() {
+    if (underlying_url_sp->GetSummaryAsCString(summary, options))
+      return true;
+    auto dynval_sp =
+        underlying_url_sp->GetDynamicValue(lldb::eDynamicCanRunTarget);
+    if (!dynval_sp)
+      return false;
+    return dynval_sp->GetSummaryAsCString(summary, options);
+  };
+  if (!try_format())
     return false;
-
   stream.PutCString(summary.c_str());
   return true;
 }
 
+bool lldb_private::formatters::swift::SwiftURL_SummaryProvider(
+    ValueObject &valobj, Stream &stream, const TypeSummaryOptions &options) {
+  static ConstString g__url("_baseURL");
+  static ConstString g__parseInfo("_parseInfo");
+  static ConstString g_urlString("urlString");
+
+  ValueObjectSP rel_str_sp(
+      valobj.GetChildAtNamePath({g__parseInfo, g_urlString}));
+  if (!rel_str_sp)
+    return false;
+
+  std::string base;
+  ValueObjectSP base_url_sp(valobj.GetChildAtNamePath({g__url}));
+  if (base_url_sp)
+    if (ValueObjectSP non_synth_valobj = base_url_sp->GetNonSyntheticValue()) {
+      const char *value = non_synth_valobj->GetValueAsCString();
+      if (value && llvm::StringRef(value) != "none")
+        if (!base_url_sp->GetSummaryAsCString(base, options))
+          return false;
+    }
+
+  std::string summary;
+  if (!rel_str_sp->GetSummaryAsCString(summary, options))
+    return false;
+
+  stream << summary;
+  if (!base.empty())
+    stream << " -- " << base;
```

Comment:
minor: if you add any commits to this PR, I think it would be nice to have a comment here saying this dash-dash format is matching the the `_SwiftURL.description` implementation.